app/lib/io/zipUtil.js

```diff
@@ -12,7 +12,31 @@ define([
     var NotImplementedError = errors.NotImplemented;
 
     var unpack = function(async, zipData, io, targetPath){
-        throw new NotImplementedError('ZIP unpack method is not yet implemented');
+        if (async)
+            throw new NotImplementedError('Asynchronous ZIP unpack method is not yet implemented');
+
+        var zip = new JSZip(zipData)
+          , files = zip.files
+          , filename
+          , file
+          , absolute_path
+          , dir_abs_path
+          ;
+
+        console.log("unpacking files: ", files);
```

Comment:
On 05/13/2015 04:10 AM, Felipe Corrêa da Silva Sanches wrote:
>
> In app/lib/io/zipUtil.js
> <https://github.com/metapolator/metapolator/pull/590#discussion_r30197904>:
>
> > @@ -12,7 +12,31 @@ define([
> >      var NotImplementedError = errors.NotImplemented;
> >  
> >      var unpack = function(async, zipData, io, targetPath){
> > -        throw new NotImplementedError('ZIP unpack method is not yet implemented');
> > +        if (async)
> > +            throw new NotImplementedError('Asynchronous ZIP unpack method is not yet implemented');
> > +
> > +        var zip = new JSZip(zipData)
> > +          , files = zip.files
> > +          , filename
> > +          , file
> > +          , absolute_path
> > +          , dir_abs_path
> > +          ;
> > +
> > +        console.log("unpacking files: ", files);
>
> How does this._log differ from console.log ?
>
The short version is that it can be controlled better. We can use log
levels etc. and log to different channels. I must admit that I did not
dig into it very deeply yet, but we added it to have more control
eventually.

Here are some docs: http://micovery.bitbucket.org/util-logging/